extensions/mssql/src/objectExplorerNodeProvider/treeNodes.ts

```diff
@@ -77,8 +78,13 @@ export abstract class TreeNode implements ITreeNode {
 	}
 
 	public async updateFileSource(connection: SqlClusterConnection): Promise<void> {
-		this.fileSource = await connection.createHdfsFileSource();
+		this._fileSource = await connection.createHdfsFileSource();
+	}
+
+	public async getFileSource(): Promise<IFileSource | undefined> {
```

Comment:
For most of the nodes this is going to be filled in properly hence the default here. But for the top level HDFS folder this isn't created until needed - so that overrides this function to handle creating the filesource.